common/assert-kubeconfig.yml

```diff
@@ -0,0 +1,67 @@
+# checks if kubeconfig pointed to correct cluster 
+- name: "Reading version info"
+  set_fact:
+    cluster_info: "{{ lookup('k8s', cluster_info='version') }}"
+  ignore_errors: true
+
+- set_fact:
+    k8s_version: "{{ cluster_info.kubernetes }}"
+  ignore_errors: true
+
+- set_fact:
+    openshift_version: "{{ cluster_info.openshift }}"
+  ignore_errors: true
+
+- when: openshift_version is not defined
+  ignore_errors: true
+  block:
+  - set_fact:
+      openshift_version_info: "{{ lookup('k8s', 
+        api_version='config.openshift.io/v1', 
+        kind='ClusterVersion',
+        resource_name='version'
+      ) }}"
+  
+  - set_fact:
+      openshift_version: "{{ openshift_version_info.status.desired.version }}"
+    when: openshift_version_info is defined and
```

Comment:
You might be interested to know that this `when` check can be performed in on one line:
```
when: openshift_version_info.get("status", {}).get("desired", {}).get("version, "") != ""
```

Syntax above may not be 100% correct, but general idea is that you can do a chain of calls like:
```
# you can run python "get" on ansible dicts. I use an empty dict as the fallback value
# except on the final get call where we will compare against an empty string
get("desiredKey", fallBackIfNotFound) 
```


The end result here is the same, so feel free to leave as-is.